python/ray/util/metrics.py

```diff
@@ -159,6 +181,8 @@ def __init__(
         tag_keys: Optional[Tuple[str, ...]] = None,
     ):
         super().__init__(name, description, tag_keys)
+        if self.discard_metric:
+            return
         self._metric = CythonCount(self._name, self._description, self._tag_keys)
```

Comment:
Consider doing the following:
```suggestion
        if self.discard_metric:
            self._metric = None
        else:
            self._metric = CythonCount(self._name, self._description, self._tag_keys)
```
Then the record method, you can check if `self._metric is not None`

Purely style, but it's a bit of an antipattern to have a "conditional field" -- better to have the field always defined with a conditional value

Reply:
Fair enough. I changed it